cmake/MPIDynamicHelper.cmake

```diff
@@ -46,27 +46,28 @@ if(NRN_ENABLE_MPI)
       # ~~~
       # Know about openmpi, mpich, ms-mpi, and a few others.
       # ~~~
```

Comment:
I wanted to know what this was used for, and in the process, I discovered `NRN_MPI_TYPE_LIST` doesn't seem to be used?

Not sure if you want to include that in your cleanup.